### DISCUSSION TITLE
Logo during exit rotating

### DISCUSSION THREAD

**Discussion by Stoppedpuma** [ppy/osu#27271] [Q&A]
Unfortunately I don't have a recording of this as I didn't have obs open at the time but when I closed the game the osu! logo and visualizer that fades out when the game closed started rotating to the right almost as if barrel roll was enabled? I did not use barrel roll during my game session if that matters.

Relevant(?) part of runtime log:
[runtime.log](https://github.com/ppy/osu/files/14343719/runtime.log)

**Comment by Stoppedpuma** [comment]
As note this is the first time I've had this issue and I can't reproduce it

**Reply by nagi-desuuu** [reply]
Maybe try double-tapping Alt+F4 very quickly while you're in song select?

**Reply by Joehuu** [reply]
Yes, that reproduces for me.

**Reply by Joehuu** [reply]
In code, it's supposed to rotate:
https://github.com/ppy/osu/blob/ef413c08f19a0743a2aa7020aa72047118828adb/osu.Game/Screens/Menu/IntroScreen.cs#L326-L331

But it doesn't work most of the time during normal circumstances.

Here's an initial investigation (using ctrl-f1):

Not rotated (bug, just quitting normally):

https://github.com/ppy/osu/assets/35318437/0d6fa66e-10c3-4199-847c-d8189ff8d117

Rotated (reproduced using alt-f4 twice):

https://github.com/ppy/osu/assets/35318437/04d1c3d5-5b7c-40c3-9adb-17885437cfc9

**Reply by Stoppedpuma** [reply]
> Maybe try double-tapping Alt+F4 very quickly while you're in song select?

If I recall correctly that was what I did when I closed the game and got it to rotate, I still haven't managed to get it to reproduce doing that but seeing has @Joehuu has that's probably it.

> In code, it's supposed to rotate

I don't remember seeing it ever do that in lazer, good to know. It does look odd to me for some reason even though it's in stable but probably since that's the first time that I've seen it happen with lazers logo.

**Reply by OwenCMYK** [reply]
Has the quit animation seriously been broken this whole time? How has nobody noticed it until now? I don't mean this at all as an insult to any players or developers, I'm just genuinely shocked that this managed to slip by everyone.

**Reply by Stoppedpuma** [reply]
@owengaming001 It's probably been noticed but no one has submitted a pr to fix it since it's not really a priority, there's probably a hundred other things that matter a lot more than a small animation when you close the game.

**Reply by OwenCMYK** [reply]
Makes sense
